### ISSUE TITLE
Provide more prescriptive guidance on multi-tenancy

### ISSUE THREAD

**Issue by sbueringer** [kubernetes-sigs/cluster-api#7636]
From the PR that we previously had for this topics:

> As a follow up to the Cluster API meeting of 2021/04/21, this PR intends to give more prescriptive guidance around the multi-tenancy contract. In particular, this documents how cluster scoped resources should work at the present time, and also in a
future release of Cluster API that drops support for older Kubernetes versions.
> https://github.com/kubernetes-sigs/cluster-api/pull/4514

There doesn't seem to be a proposal in core Cluster API and the book only has this short documentation: https://cluster-api.sigs.k8s.io/developer/architecture/controllers/multi-tenancy.html
> [Multi tenancy](https://cluster-api.sigs.k8s.io/developer/architecture/controllers/multi-tenancy.html#multi-tenancy)
Multi tenancy in Cluster API defines the capability of an infrastructure provider to manage different credentials, each one of them corresponding to an infrastructure tenant.
> [Contract](https://cluster-api.sigs.k8s.io/developer/architecture/controllers/multi-tenancy.html#contract)
In order to support multi tenancy, the following rule applies:
>
> Infrastructure providers MUST be able to manage different sets of credentials (if any)
> Providers SHOULD deploy and run any kind of webhook (validation, admission, conversion) following Cluster API codebase best practices for the same release.
> Providers MUST create and publish a {type}-component.yaml accordingly.

#4514 was a lot more concrete than our current documentation. I'm not sure what the current state on multi-tenancy implementations across providers is and what the starting point for a new PR would be.

/kind documentation
[One or more /area label. See https://github.com/kubernetes-sigs/cluster-api/labels?q=area for the list of labels]

**Comment by sbueringer** [comment]
cc @randomvariable @fabriziopandini @CecileRobertMichon @richardcase 
In case you want to add something. Just thought it's better to at least track this in an issue vs a closed PR.

**Comment by fabriziopandini** [comment]
/triage accepted
/help
It would be great to revive https://github.com/kubernetes-sigs/cluster-api/pull/4514, but at the same time, I would like provider's teams to take ownership of defining a common baseline that the users can rely on across different providers and make it a formal contract.
If it could help. let's bring up this topic in the weekly meeting ...

**Comment by sbueringer** [comment]
> /triage accepted /help It would be great to revive #4514, but at the same time, I would like provider's teams to take ownership of defining a common baseline that the users can rely on across different providers and make it a formal contract. If it could help. let's bring up this topic in the weekly meeting ...

I agree. I just had it on my list of things that we just lost track of. I definitely think this should be discussed with providers if we want to get to more conformity that we currently have documented.

Added it to the agenda for this wekk.

**Comment by sbueringer** [comment]
@randomvariable just fyi :)
